### PR TITLE
Calib all symbols

### PR DESCRIPTION
**Pull request by lawhead**
# Overview

Refactored stimuli generation to ensure that all symbols are shown in a calibration.

## Ticket

https://www.pivotaltracker.com/story/show/185235958

## Contributions

- Refactored stimuli generation to be break large functions into smaller units that can be more easily tested and combined.
- Added new criteria for targets (frequency)
- Added test coverage
- Added utility functions for analyzing the statistical properties of symbols in an inquiry schedule.

## Test

- Ran unit tests, including those to ensure that previous criteria are still being met
- Ran a sample calibration to ensure the refactored names were properly integrated in the tasks.

### CODE REVIEW

File: bcipy/helpers/stimuli.py [hunk 1]
```diff
@@ -517,95 +527,293 @@ def calibration_inquiry_generator(
...
     """
-    assert len(timing) == 3, "timing must include values for [target, fixation, stimuli]"
-    target_indexes = []
-    no_target = None
-
-    if (target_positions == target_positions.DISTRIBUTED):
-        target_indexes = distributed_target_positions(stim_number, stim_length, nontarget_inquiries)
+    if timing is None:
+        timing = [0.5, 1, 0.2]
+    if color is None:
```
Comment (tab-cmd) [inline bcipy/helpers/stimuli.py:532]: I'd be okay making these required inputs. I don't think we gain much from having these defaults.

**Review by tab-cmd — APPROVED** [review]
This is a really clean addition! Thanks for the PR, Matt!

File: bcipy/helpers/stimuli.py [hunk 2]
```diff
@@ -480,21 +481,30 @@ def best_case_rsvp_inq_gen(alp: list,
...
-        color: List[str] = ['green', 'red', 'white'],
-        stim_number: int = 10,
-        stim_length: int = 10,
+        color: List[str] = None,
+        inquiry_count: int = 100,
+        stim_per_inquiry: int = 10,
         stim_order: StimuliOrder = StimuliOrder.RANDOM,
         target_positions: TargetPositions = TargetPositions.RANDOM,
-        nontarget_inquiries: int = 10,
+        percentage_without_target: int = 10,
```
Comment (celikbasak) [inline bcipy/helpers/stimuli.py]: Based on our previous discussions maybe we can scrap this number and set the default as 0?

**Review by celikbasak — APPROVED** [review]
This is a great update, thank you so much Matt!

File: bcipy/helpers/stimuli.py (hunk 1 above)
Comment (lawhead) [reply bcipy/helpers/stimuli.py:532]: I think the main advantage is in testing and demos when we primarily care about the symbols.

**Pull request merged**